app/controllers/organizations_controller.rb

```diff
@@ -49,6 +49,21 @@ def update
     end
   end
 
+  def destroy
+    organization = Organization.find_by(id: params[:id])
+    authorize organization
+    if destroyable_organization?(organization)
```

Comment:
I like this, though I would put it in the model as `organization.destroyable?` or even better, put it as a condition for the authorization, `:destroy` and remove the if. After all that's what we're saying to the user: you're not authorized to destroy this organization because this and that.

Reply:
Hmm so if I put the condition in the authorization, would I rescue the error? For example:
```ruby
def destroy
    organization = Organization.find_by(id: params[:id])
    authorize organization
    organization.destroy
    current_user.touch(:organization_info_updated_at)
    CacheBuster.bust_user(current_user)
    flash[:settings_notice] = "Your organization: \"#{organization.name}\" was successfully deleted."
    redirect_to user_settings_path(:organization)
  rescue Pundit::NotAuthorizedError
    flash[:settings_notice] = "Your organization was not deleted; you must be an admin,
       the only member in the organization, and  no articles connected to the organization."
    redirect_to user_settings_path(:organization, id: organization.id)
  end
```

That seems a bit counterintuitive but maybe I'm approaching it incorrectly?